samples/extensions/shader_debugprintf/shader_debugprintf.cpp

```diff
@@ -375,6 +391,13 @@ bool ShaderDebugPrintf::prepare(const vkb::ApplicationOptions &options)
 		return false;
 	}
 
+	// Register debug utils callback here vs in ShaderDebugPrintf::create_instance() so it works with both override and layer settings
```

Comment:
Can you elaborate on this comment and also check if the readme for this sample needs to be adjusted?

Reply:
There are two code paths now for enablement of the validation layer and the debug printf feature of the validation layer:  a) the existing approach that uses a custom override for `create_instance()`, and b) a new approach that enables the validation layer using a simple `get_validation_layers()` override (for release builds) combined with the layer settings method of enabling the debug printf feature of the validation layer.  The latter does not use the custom `create_instance()` override code, but instead calls the standard `VulkanSample::create_instance()` method if `VK_EXT_layer_settings` is available at runtime.

The registration of the debug utils callback has to happen after either approach has finished, and the best location for that is at the beginning of `ShaderDebugPrintf::prepare()`.